foqus_lib/framework/uq/Common.py

```diff
@@ -66,7 +66,6 @@ def showError(self, error, out=None):
                 )
                 if out is not None:
                     msgBox.setDetailedText(out)
-                msgBox.exec_()
```

Comment:
Does removing this potentially hide error messages generated by non-Plugin tools? I'm wondering if we need to keep this

Reply:
This is a valid point. Unfortunately, using modal dialogs such as `QMessageBox` through the `exec_()` method causes all sorts of issues with automated GUI testing. More specifically this has to do with Qt's event loop: basically calling `exec_()` halts the execution of the main event loop in which the main window is running, and starts a separate event loop for the modal dialog. Once the modal dialog is dismissed, the main event loop resumes. This is fine during interactive use; however, the infrastructure used for automated testing (the "qtbot") is also running in the main event loop, and is therefore "trapped" whenever a modal dialog is created with `exec_()`. I had implemented a workaround that seemed to be working, however this started causing issues when adding the `test_ml_ai.py` file.

Long story short, getting rid of this particular modal is a temporary fix to get these set of tests to work on CI. I'm already working on a more robust solution that will encapsulate user notifications with a PyQt signal, that can be connected to display a modal dialog for interactive use, or inspected directly (with the modal dialogs turned off) for automated testing.